### PR TITLE
Change Documentation to YARD format

### PR DESCRIPTION
**Pull request by dlitvakb**
Changes docs to YARD
Fixes Rubocop violations
Adds Locale Documentation on README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,6 +100,27 @@ Dynamic entries will have getter classes for the fields and do type conversions
 
 The `:auto` mode will fetch all content types on initialization. If you want to do it by hand later, you will need to set the option to `:manual` and call `client.update_dynamic_entry_cache!` to initialize all dynamic entries.
 
+#### Using different locales
+
+Entries can have multiple locales, by default, the client only fetches the entry with only it's default locale.
```
Comment (neonichu) [inline README.md]: its

File: README.md [hunk 2]
```diff
@@ -100,6 +100,27 @@ Dynamic entries will have getter classes for the fields and do type conversions
...
+Contentful Delivery API also allows to fetch all locales, you can do so by doing:
+
+```ruby
+entries = client.entries(content_type: 'cat', locale: '*')
+
+# assuming the entry has a field called name
+my_spanish_name = entries.first.fields('es-AR')[:name]
+```
+
+When requesting multiple locales, the object accessor shortcuts only work for the default locale
```
Comment (neonichu) [inline README.md]: .

File: lib/contentful/client.rb [hunk 3]
```diff
@@ -43,6 +45,24 @@ def self.get_http(url, query, headers = {}, proxy = {})
       end
     end
 
+    # @param [Hash] given_configuration
+    # @option given_configuration [String] :space Required
+    # @option given_configuration [String] :access_token Required
+    # @option given_configuration [String] :api_url
```
Comment (neonichu) [inline lib/contentful/client.rb]: Might be worth mentioning that by changing this, people can access the CPA.

File: lib/contentful/client.rb [hunk 4]
```diff
@@ -43,6 +45,24 @@ def self.get_http(url, query, headers = {}, proxy = {})
       end
     end
 
+    # @param [Hash] given_configuration
```
Comment (neonichu) [inline lib/contentful/client.rb]: Might make sense to actually take stuff from <https://github.com/contentful/contentful.rb#client-configuration-options> and put it here.

File: README.md (hunk 1 above)
Comment (dlitvakb) [inline README.md]: It uses the client's specified default locale

**Pull request merged**